game/src/select_cards.rs

```diff
@@ -49,13 +52,343 @@ fn get_crib_cards(hand: &[Card], held_cards: &[Card]) -> Vec<Card> {
+ *  player knows this is a normal thing to do, so by playing a 2 you imply you also have a 3.  So maybe the right card to play is a 4 so that the
+ *  apponent plays the 3 to get the run (2, 4, 3) and 3 points - and then bam! -- you play an Ace to get 4 points.  and so on.
+ *
+ *  my strategy for this is to implement it the way I play the game.  by its nature it is a bunch of "if this then that" calls, so this function
+ *  tends to be very long and very experimental.  I have added logic in the past and then tested it by playing millions (literally) of games of
+ *  one algorithm against the other to see if it really works (it takes about 20 seconds to run a million games on a beefy PC...)
+ *
+ *
+ */
+pub fn get_next_counted_card(
+    counted_cards: Vec<Card>,
+    mut cards_left: Vec<Card>, // needs to be mut because we .sort() it
+) -> Result<Card, Box<dyn Error>> {
+    let current_count: i32 = counted_cards.iter().map(|c| c.value).sum::<i32>();
+    //
+    //  if you only have one card left, play it if you can
+    if cards_left.len() == 1 {
+        if current_count + cards_left[0].value <= 31 {
+            return Ok(cards_left[0]);
+        } else {
+            return Err("Go".into());
+        }
+    }
 
-mod tests {
-    // import names from outer scope
+    cards_left.sort(); // this is a sort by rank
+
+    //
+    //  here is the first strategic decision - this algorithm will take points if they are available.  for example, when
+    //  humans play, they might decide to defer points (say a pair) in order to get the opponent to start a run.  e.g. if cards_played is something like
+    //  a 2, then it would be common for that person to also have a 3.  they played the 2 hoping that a card of value 10 would be played and then they
+    //  could play a 3 - 15 for 2 points!.  This algorithm will always play a 2 to pair the 2 if it is possible.  Another strategy would be to play a 4
+    //  or an Ace (*never* a 3 - because you don't want the count to be 5 because of how many cards are available with value=10).  in this case, the opponent
+    //  would play the 3 and get a run of 3 for 3 points.  But then you could build on the run and get 4 points (assuming you had the right cards)
+    //
+    //  one of the things I want to explore is defeating the "get a 15" strategy above. the other common strategy (which we use below) is to start with a card
+    //  that you have a pair...the hope is the apponent pairs the card, so you can get 3 of a kind (e.g. play a 4, opponent plays a 4, you play your second 4)
+    //  would it be better to not pair the card and give up the two points?  over the course of the game, stopping 3 of a kind might be worth it.
+    //
+    //  this is an area ripe for innovation!
+    //
+
+    //
+    //  Note: I need the card to play for the max score - i tried to do this in a variety of ways using map/fold and I could get the max score, but the card
```

Comment:
You could accumulate to a tuple of `(i32, Card)` with map/fold, so you don't lose the card. I think that _would_ be cleaner.